### ISSUE TITLE
New game definitions created on Steam Deck not being recreated on PC installation

### ISSUE THREAD

**Issue by dc-grimm** [DavidDeSimone/OpenCloudSaves#80]
Hi David,

I created a definition for syncing god of war ps2 and some other emulated games on the Steam Deck.

When Syncing on pc these don't get created.

**Comment by DavidDeSimone** [comment]
This is an area I am trying to improve in 0.18 - try opening the app on windows, waiting a few seconds, and then restarting the app on windows.

The issue if that while we do sync game definitions across the platforms, we don't refresh the UI when that operation is complete, which can lead to user confusion. 

it is also possible there is a bug in the sync logic, but I'd like to start with that step first.

**Comment by dc-grimm** [comment]
ps. I tried downloading the user_overrides.json from the cloud and replaced it in %appdata%\OpenCloudSave but that seems to be the default game list without updates.

**Comment by DavidDeSimone** [comment]
> ps. I tried downloading the user_overrides.json from the cloud and replaced it in %appdata%\OpenCloudSave but that seems to be the default game list without updates.

Hmmmm, so that means that your user_overrides didn't update to the cloud. I'll investigate this more. For now, you may need to recreate the definition across your devices - sorry about that!

**Comment by dc-grimm** [comment]
Update: After a few syncs on the Deck and manual adding it to the PC one its now showing the new entries.

It seems that the sync on the pc takes way longer than on the deck though

**Comment by dc-grimm** [comment]
Okay. It was still busy simulating so I killed it.
I think I broke it
![image](https://user-images.githubusercontent.com/15102883/224146431-640fd039-491d-4045-a4cb-6d85adb91310.png)

**Comment by DavidDeSimone** [comment]
yeah good catch - it may have been defaulting to a "root" path so it was syncing everything under the sun. That is a bad footgun, and I'll add some more protections around that. 

Sorry you are having these issues! I'll work to try and correct some of these rough edges for next release.

**Comment by dc-grimm** [comment]
Na man. No problem.
Its all good. Having this tool is already a win, and helping to make it better makes it even better.
Unfortunately I don't have the skills you are employing here or I would have loved to help more. 

Btw. I had trouble installing it on pc. It just popped up a cmd window and closed without options.

Not sure how it finally got installed to be honest. I opened the installer with zip and ran the application from extracted and after that I found it installed under program files. So that might be something else to also look at in future.

**Comment by DavidDeSimone** [comment]
For some guardrails on large syncs, we will now have this:

<img width="803" alt="Screenshot 2023-03-09 at 8 07 16 PM" src="https://user-images.githubusercontent.com/7245174/224220967-5b384e0a-ca53-4e20-99e0-9f442b4abc5f.png">
<img width="803" alt="Screenshot 2023-03-09 at 8 07 08 PM" src="https://user-images.githubusercontent.com/7245174/224220974-30f54061-8223-41e5-a3fd-9b944a2f68ef.png">

I've also put a cap on log files to prevent them from blowing up to a huge size.

**Comment by DavidDeSimone** [comment]
So I figured out the root cause - even though I was syncing user_overrides.json, I was writing to it right before the cloud sync operation. So user_override would always have a newer timestamp compared to what was on the cloud, and it would not sync (and often get in weird states).

**Comment by gingerr** [comment]
Stumbled upon the same problem when trying out OpenCloudSaves to sync Zelda BotW between Steam Deck and PC. I will attempt next to sync the `user_overrides.json` manually once i localized it on the steam deck.

Do you plan to build and distribute a new OpenCloudSaves version which includes #104 ?

Thank you for OpenCloudSaves!
